### PR TITLE
[FLINK-33818] Implement restore tests for WindowDeduplicate node

### PR DESCRIPTION
**Pull request by bvarghese1**
## What is the purpose of the change

*Add restore tests for WindowDeduplicate node*

## Verifying this change
This change added tests and can be verified as follows:

- Added restore tests for WindowDeduplicate node which verifies the generated compiled plan with the saved compiled plan

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (no)
  - The serializers: (no)
  - The runtime per-record code paths (performance sensitive): (no)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: (no)
  - The S3 file system connector: (no)

## Documentation

  - Does this pull request introduce a new feature? (no)
  - If yes, how is the feature documented? (not applicable)

### CODE REVIEW

File: flink-table/flink-table-planner/src/test/java/org/apache/flink/table/planner/plan/nodes/exec/stream/WindowDeduplicateTestPrograms.java [hunk 1]
```diff
@@ -0,0 +1,287 @@
...
+
+    static final String HOP_TVF =
+            "TABLE(HOP(TABLE bid_t, DESCRIPTOR(bid_time), INTERVAL '5' SECOND, INTERVAL '10' SECOND))";
+
+    static final String CUMULATIVE_TVF =
+            "TABLE(CUMULATE(TABLE bid_t, DESCRIPTOR(bid_time), INTERVAL '5' SECOND, INTERVAL '10' SECOND))";
+
+    static final String ONE_ROW = "row_num <= 1";
+
+    static final String N_ROWS = "row_num < 3";
```
Comment (dawidwys) [inline flink-table/flink-table-planner/src/test/java/org/apache/flink/table/planner/plan/nodes/exec/stream/WindowDeduplicateTestPrograms.java:85]: Is this still converted to `WindowDeduplicate`? I think it tests `WindowRank`, doesn't it? The json plans seem to confirm as I can not find the `WindowDeduplicate` node there.

**Review by dawidwys — CHANGES_REQUESTED** [review]
I like the PR! The only suggestion would be to make sure we group by the node that is actually tested.

File: flink-table/flink-table-planner/src/test/java/org/apache/flink/table/planner/plan/nodes/exec/stream/WindowDeduplicateTestPrograms.java (hunk 1 above)
Comment (bvarghese1) [reply flink-table/flink-table-planner/src/test/java/org/apache/flink/table/planner/plan/nodes/exec/stream/WindowDeduplicateTestPrograms.java:85]: Yes, my bad. Will move this to the WindowRank PR.

Comment (dawidwys) [reply flink-table/flink-table-planner/src/test/java/org/apache/flink/table/planner/plan/nodes/exec/stream/WindowDeduplicateTestPrograms.java]: Thank you for the update! LGTM now

**Review by dawidwys — APPROVED** [review]
LGTM

**Pull request merged**